3-UsingAPIs/Week2/assignment/ex2-pokemonApp/index.js

```diff
@@ -21,18 +23,72 @@ Use async/await and try/catch to handle promises.
 Try and avoid using global variables. As much as possible, try and use function 
 parameters and return values to pass data back and forth.
 ------------------------------------------------------------------------------*/
-function fetchData(/* TODO parameter(s) go here */) {
-  // TODO complete this function
+async function fetchData(url) {
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error(`HTTP ${response.status}`);
+    }
+    const data = await response.json();
+    return data;
```

Comment:
Compared to ex1, while you can still get rid of the `data` variable, you can no longer get rid of the `await` keyword if you want to catch a potential error thrown by `response.json()` in the local `catch` block. The `await` keyword will wait for the promise to settle and throw an error if the promise from `response.json()` is rejected. This error can then be caught by the local `catch` block.

```suggestion
    return await response.json();
```

If you were to leave out the `await` too, like so: `return response.json()`, then `fetchData()` would return the promise from `response.json()` directly. In case `response.json()` throws an error (e.g. if no valid json was sent by the server), the local `catch` block would not be triggered, however the error would still be caught by the `try/catch` block in the calling function.